src/neuroglancer/sliceview/backend_chunk_decoders/compresso.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * @license
+ * Copyright 2021 William Silvermsith.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {TypedArray} from 'neuroglancer/util/array';
+import {DataType} from 'neuroglancer/sliceview/base';
+import {postProcessRawData} from 'neuroglancer/sliceview/backend_chunk_decoders/postprocess';
+import {VolumeChunk} from 'neuroglancer/sliceview/volume/backend';
+import {CancellationToken} from 'neuroglancer/util/cancellation';
+import {decodeCompresso} from 'neuroglancer/async_computation/decode_compresso_request';
+import {requestAsyncComputation} from 'neuroglancer/async_computation/request';
+
+export async function decodeCompressoChunk(
+    chunk: VolumeChunk, cancellationToken: CancellationToken, response: ArrayBuffer) {
+  
+  let image : TypedArray = await requestAsyncComputation(
+    decodeCompresso, cancellationToken, [response], new Uint8Array(response)
+  );
+
+  const dtype = chunk.source!.spec.dataType || DataType.UINT8;
+
+  // uint8 is already set correctly
```

Comment:
Isn't this redundant with the code you have in `read_image` for setting the TypedArray type?

Reply:
Yes, now that you mention it, this could be a two-liner. I combined this with your suggestion on try/finally.